### ISSUE TITLE
D2.1 Fix and consolidate database scripts

### ISSUE THREAD

**Issue by obrienben** [DIA-NZ/webcurator#27]
Fix and test database scripts: 
- all DML/DDL files that are needed to create the 1.7.x WCT database from scratch.
- all DML/DDL files that are needed to upgrade from 1.6.3 to 1.7.x

Note that there are separate sets of scripts for each of the platforms MySQL, PostgreSQL, Oracle and H2.

**Comment by obrienben** [comment]
I wonder if we should think about tidying up the db folder @hannakoppelaar as well? Separating the scripts to install a fresh copy of the latest version, from the legacy ones. For instance something like,, 

- db/
  - latest/
    - setup/
    - sql/
    - upgrade/
  - legacy/
    - setup/
    - sql/
    - upgrade/

**Comment by hannakoppelaar** [comment]
@obrienben Good idea. And perhaps we can then drop the version numbers from the file names (e.g. wct-schema-grants-**1_6_1**-mysql.sql) as well?

**Comment by hannakoppelaar** [comment]
What I've done with the mysql and postgresql scripts in the sql directory is:

* take the latest version of those;
* add the alterations from the latest upgrade scripts (from 1.6 to 1.6.1) to wct-schema-<db>.sql, not as ALTER statements, but by modifying the existing CREATE statements;
* likewise, add the column definitions introduced by ALTER statements in wct-schema-<db>.sql (below the comment "WCT 1.6 upgrade") to the existing CREATE statements (in the same file); this is just a tidying operation.

I'm using these throwaway scripts to setup the databases:

MySQL:

#!/bin/sh

export MYSQL_PWD=password

WCT_DB_DIR=/usr/local/tmp/wct/wct-core/db/latest
LOGFILE=/tmp/wct-db-setup.log

if [ -e $LOGFILE ]
then
        rm $LOGFILE
fi

cat $WCT_DB_DIR/setup/mysql-create-wct.sql | mysql -u root >> $LOGFILE 2>&1
cat $WCT_DB_DIR/sql/wct-schema-mysql.sql | mysql -u root -D DB_WCT >> $LOGFILE 2>&1
cat $WCT_DB_DIR/sql/wct-schema-grants-mysql.sql | mysql -u root -D DB_WCT >> $LOGFILE 2>&1
cat $WCT_DB_DIR/sql/wct-indexes-mysql.sql | mysql -u root -D DB_WCT >> $LOGFILE 2>&1
cat $WCT_DB_DIR/sql/wct-mysql-bootstrap.sql | mysql -u root -D DB_WCT >> $LOGFILE 2>&1
cat $WCT_DB_DIR/sql/wct-qa-data-mysql.sql | mysql -u root -D DB_WCT >> $LOGFILE 2>&1


PostgreSQL:

#!/bin/sh

export PGPASSWORD=password

WCT_DB_DIR=/usr/local/tmp/wct/wct-core/db/latest
LOGFILE=/tmp/wct-db-setup.log

if [ -e $LOGFILE ]
then
        rm $LOGFILE
fi

cat $WCT_DB_DIR/setup/postgres-create-wct.sql | psql -U postgres >> $LOGFILE 2>&1
cat $WCT_DB_DIR/sql/wct-schema-postgresql.sql | psql -U postgres --dbname=Dwct >> $LOGFILE 2>&1
cat $WCT_DB_DIR/sql/wct-schema-grants.sql | psql -U postgres --dbname=Dwct >> $LOGFILE 2>&1
cat $WCT_DB_DIR/sql/wct-indexes-postgresql.sql | psql -U postgres --dbname=Dwct >> $LOGFILE 2>&1
cat $WCT_DB_DIR/sql/wct-postgres-bootstrap.sql | psql -U postgres --dbname=Dwct >> $LOGFILE 2>&1
cat $WCT_DB_DIR/sql/wct-qa-data-postgres.sql | psql -U postgres --dbname=Dwct >> $LOGFILE 2>&1

**Comment by obrienben** [comment]
@hannakoppelaar I'm just going through the oracle upgrade script - which changes did you make to wct-schema-mysql.sql to replace the following lines in upgrade-mysql-1_6-to-1_6_1.sql?

- alter table DB_WCT.ROLE_PRIVILEGE drop foreign key FK_PRIV_ROLE_OID; **(line 14)**
- delete from DB_WCT.ROLE_PRIVILEGE where PRV_ROLE_OID is null; **(line 18)**
- alter table DB_WCT.ROLE_PRIVILEGE add constraint FK_PRIV_ROLE_OID foreign key (PRV_ROLE_OID) references DB_WCT.WCTROLE (ROL_OID) on delete cascade; **(line 19)**

**Comment by obrienben** [comment]
@hannakoppelaar I've committed the oracle changes, removed the upgrade folder from latest, and tested H2 (via Jetty) is still working. We've also tested your mysql scripts. 

- Is there anything else needed? 
- Is Rene happy with the changes?
- I was wondering if any upgrade script work for moving 1.6.3 to 1.7.x should be part of D1.4 so we can merge this one into master? 

If I you think it's done, assign me the PR and I'll try and test the postgres scripts
